pandas/core/indexes/base.py

```diff
@@ -4640,7 +4640,7 @@ def join(
         return_indexers: Literal[True],
         sort: bool = ...,
     ) -> tuple[Index, npt.NDArray[np.intp] | None, npt.NDArray[np.intp] | None]:
-        ...
+        raise NotImplementedError
```

Comment:
Since this seems to be an issue with pylint https://github.com/PyCQA/astroid/issues/1015, I don't think it's worth the work-around

Reply:
Makes sense, I'll close the PR.